source/MaterialXRender/ImageHandler.h

```diff
@@ -86,6 +88,25 @@ class MX_RENDER_API ImageSamplingProperties
     Color4 defaultColor = { 0.0f, 0.0f, 0.0f, 1.0f };
 };
 
+/// @struct ImageSamplingKeyHasher
+/// Class used for hashing ImageSamplingProperties in an unordered_map
+struct MX_RENDER_API ImageSamplingKeyHasher
+{
+  std::size_t operator()(const ImageSamplingProperties& k) const
+  {
+    using std::size_t;
+    using std::hash;
+    using std::string;
+
+    return
+      (std::size_t)(k.enableMipmaps)                         + // 1 bit
+      (((std::size_t)k.filterType & 3)                 << 1) + // 2 bit
+      (((std::size_t)((int)k.uaddressMode + 1)    & 7) << 3) + // 3 bit
+      (((std::size_t)((int)k.vaddressMode + 1)    & 7) << 6) + // 3 bit
+      (((std::size_t)((int)k.defaultColor[0] + 1)    ) << 9) ;
+  }
```

Comment:
When I run PVS-Studio static analysis on this pull request, it points out that this function performs bit shifts on signed integers and then casts the results to unsigned integers, which incurs some potential overflow risks.

What would you think about the following alternative implementation?

```
struct MX_RENDER_API ImageSamplingKeyHasher
{
    size_t operator()(const ImageSamplingProperties& k) const
    {
        return (size_t) k.enableMipmaps                   + // 1 bit
               (((size_t) k.filterType         & 3) << 1) + // 2 bit
               ((((size_t) k.uaddressMode + 1) & 7) << 3) + // 3 bit
               ((((size_t) k.vaddressMode + 1) & 7) << 6) + // 3 bit
               ((((size_t) k.defaultColor[0] + 1))  << 9) ;
    }
};
```

Reply:
Thanks @jstone-lucasfilm . That looks great

Reply:
I took this change and submitted it to latest

Reply:
Great, thanks @Morteeza!